main/src/ca/uwaterloo/flix/tools/pkg/MavenPackageManager.scala

```diff
@@ -15,25 +15,67 @@
  */
 package ca.uwaterloo.flix.tools.pkg
 
+import ca.uwaterloo.flix.tools.pkg.Dependency.MavenDependency
 import ca.uwaterloo.flix.util.Result
+import ca.uwaterloo.flix.util.Result.{Err, Ok, ToOk}
 
 import java.io.PrintStream
 import coursier._
 
 object MavenPackageManager {
 
-  def installDeps(manifest: Manifest)(implicit out: PrintStream): Result[Unit, ManifestError] = {
+  val scalaVersion = "2.13"
 
+  /**
+    * Installs all MavenDependencies for a Manifest,
+    * including transitive dependencies using coursier.
+    */
+  def installAll(manifest: Manifest)(implicit out: PrintStream): Result[List[String], PackageError] = {
+    val depStrings = getMavenDependencyStrings(manifest)
 
-    val resolution = Resolve()
-      .addDependencies(dep"org.tpolecat:doobie-core_2.12:0.6.0")
-      .run()
-    println(resolution)
+    createCoursierDependencies(depStrings).flatMap { deps =>
+      val installed = try {
+        val res = deps.foldLeft(Resolve())((res, dep) => res.addDependencies(dep))
+        val resolution = res.run()
 
-    ???
+        val fetch = resolution.dependencies.foldLeft(Fetch())((f, dep) => {
+          out.println(s"Installing ${dep.module.toString()}");
+          f.addDependencies(dep)
+        })
+        fetch.run()
 
+        resolution.dependencies.map(dep => dep.module.toString()).toList
+      } catch {
+        case e: Exception =>
+          val message = e.getMessage.replaceAll("[^a-zA-Z0-9:. ]", "/").split('/').apply(0)
+          return Err(PackageError.CoursierError(s"Error in downloading Maven dependency: $message"))
+      }
+
+      Ok(installed)
+    }
+  }
+
+  /**
+    * Finds the MavenDependencies for a Manifest
+    * and converts them to Strings.
+    */
+  private def getMavenDependencyStrings(manifest: Manifest): List[String] = {
+    manifest.dependencies.collect {
+      case dep: MavenDependency => dep
+    }.map(dep => s"${dep.groupId}:${dep.artifactId}:${dep.version.toString}")
   }
 
-  private def installArtifact(groupId: String, artifactId: String, version: String)(implicit out: PrintStream): Result[Unit, ManifestError] = ???
+  /**
+    * Creates Coursier dependencies from a list of Strings
+    */
+  private def createCoursierDependencies(depStrings: List[String]): Result[List[Dependency], PackageError] = {
```

Comment:
Two things:

1. In FP you almost never write functions `List[a] -> List[b]` because then you might as well write `a -> b` and use `map`.
2. Since we already checked the format of each string, maybe we should just throw `InternalCompilerException` and return the Dependency. I think I would do that now, when reflecting on it.

Reply:
There are some things that Coursier catches that are not checked in the ManifestParser, e.g. characters that are not allowed in a dependency name

Reply:
OK, we should add those checks :) Maybe we can just restrict ourselves to a-zA-Z0-9 for all package names.